app/domain/authentication/authn_k8s/extract_k8s_resource_restrictions.rb

```diff
@@ -0,0 +1,71 @@
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+
+    ExtractK8sResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions: ResourceRestrictions::ExtractResourceRestrictions.new,
+        resource_restrictions_class:   ResourceRestrictions::ResourceRestrictions,
+        logger:                        Rails.logger
+      },
+      inputs:   %i(authenticator_name service_id role_name account)
+    ) do
+
+      def call
+        extract_resource_restrictions
+      end
+
+      private
+
+      # Extracts resource restrictions for k8s hosts. It can be defined in the
+      # annotations or as part of the host ID itself.
+      # If the restrictions are in the annotations, then they will not be
+      # extracted from the host ID.
+      def extract_resource_restrictions
+        get_restrictions_from_annotations = @extract_resource_restrictions.call(
+          authenticator_name: @authenticator_name,
+          service_id:         @service_id,
+          role_name:          @role_name,
+          account:            @account
+        )
+
+        # Container name annotation is not a restriction, and needs to be excluded.
+        resource_restrictions = get_restrictions_from_annotations.except(Restrictions::AUTHENTICATION_CONTAINER_NAME)
```

Comment:
do you think it will make more sense to move this logic into `ExtractResourceRestrictions`? if the container name is not really a restriction then the `ExtractResourceRestrictions` class can treat it that way without the need to exclude it in the calling class(es).

Reply:
This is not a regular calling class, this is an expansion of `ExtractResourceRestrictions`.
As I see it, if I had inheritance relation between these classes I would put this code in the overriding function.
That's why it's here.